### ISSUE TITLE
pre middleware

### ISSUE THREAD

**Issue by tblobaum** [briankircho/mongoose-schema-extend#7]
```js

new ExtendedModel().save()
// pre save middleware placed on ExtendedModel.schema executes

ExtendedModel.findOne(function (e, doc) {

  console.log(doc instanceof ExtendedModel)
  // => true

  doc.save()
  // pre save middleware placed on ExtendedModel.schema does NOT execute
  // only the inherited schema's middleware works

})

```

**Comment by SerjoPepper** [comment]
+1

**Comment by sieira** [comment]
Working on it:
https://github.com/briankircho/mongoose-schema-extend/pull/35

**Comment by sieira** [comment]
Well, this solves it

https://github.com/briankircho/mongoose-schema-extend/pull/36

Unfortunately, it forces the user to include the --harmony-proxies flag.

#35 solves it as well, but not so well, and anyway O.o will soon be deprecated, I will merge the proxies approach in a couple of days if no-one has a better solution.

**Comment by sieira** [comment]
This is solved in 2.0.1
